e2e/pod.go

```diff
@@ -623,3 +624,83 @@ func verifySeLinuxMountOption(
 
 	return nil
 }
+
+func verifyReadAffinity(
+	f *framework.Framework,
+	pvcPath, appPath, daemonSetName, cn, ns string,
+) error {
+	readFromReplicaOption := "read_from_replica=localize"
+	expectedCrushLocationValues := map[string]string{
+		strings.Split(crushLocationRegionLabel, "/")[1]: crushLocationRegionValue,
+		strings.Split(crushLocationZoneLabel, "/")[1]:   crushLocationZoneValue,
+	}
+	actualCrushLocationValues := make(map[string]string)
+
+	// create PVC
+	pvc, err := loadPVC(pvcPath)
+	if err != nil {
+		return fmt.Errorf("failed to load pvc: %w", err)
+	}
+	pvc.Namespace = f.UniqueName
+	err = createPVCAndvalidatePV(f.ClientSet, pvc, deployTimeout)
+	if err != nil {
+		return fmt.Errorf("failed to create PVC: %w", err)
+	}
+	app, err := loadApp(appPath)
+	if err != nil {
+		return fmt.Errorf("failed to load application: %w", err)
+	}
+	app.Namespace = f.UniqueName
+	err = createApp(f.ClientSet, app, deployTimeout)
+	if err != nil {
+		return fmt.Errorf("failed to create application: %w", err)
+	}
+
+	pod, err := f.ClientSet.CoreV1().Pods(f.UniqueName).Get(context.TODO(), app.Name, metav1.GetOptions{})
+	if err != nil {
+		framework.Logf("Error occurred getting pod %s in namespace %s", app.Name, f.UniqueName)
+
+		return fmt.Errorf("failed to get pod: %w", err)
+	}
+
+	nodepluginPodName, err := getDaemonsetPodOnNode(f, daemonSetName, pod.Spec.NodeName, ns)
+	if err != nil {
+		return fmt.Errorf("failed to get daemonset pod on node: %w", err)
+	}
+	logs, err := frameworkPod.GetPodLogs(context.TODO(), f.ClientSet, ns, nodepluginPodName, cn)
+	if err != nil {
+		return fmt.Errorf("failed to get pod logs from container %s/%s/%s : %w", ns, nodepluginPodName, cn, err)
+	}
+
+	if !strings.Contains(logs, readFromReplicaOption) {
+		return fmt.Errorf("option %s not found in logs: %s", readFromReplicaOption, logs)
+	}
```

Comment:
IMO this is not the best solution because if we log this option anywhere but we don't pass it to map/mount option this test will pass. For RBD how about checking the map options in the device config file? like `/sys/devices/rbd/0/config_info` this ensures that map option is applied for the mapped rbd image.

Reply:
@Madhu-1,
looks like I can’t read `/sys/devices/rbd/0/config_info`  file
```
root@csi-rbd-demo-pod:/sys/devices/rbd/0# ls -l config_info 
-r-------- 1 root root 4096 Sep 21 05:35 config_info
root@csi-rbd-demo-pod:/sys/devices/rbd/0# cat config_info 
cat: config_info: Operation not permitted
root@csi-rbd-demo-pod:/sys/devices/rbd/0# 
```

Reply:
@iPraveenParihar not from the demo pod but check it from the csi-rbdplugin container

Reply:
I got it, I was doing exec in the pod.

From csi-rbdplugin-xxx pod csi-rbdplugin container
```
[root@m1 0]# cat config_info 
10.99.51.241:6789 name=admin,key=client.admin,crush_location=region:east|zone:east-zone1,read_from_replica=localize replicapool csi-vol-134dadfd-4af1-4026-b79f-8fd265974a2b -
```